internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -546,9 +547,14 @@ func (r *PreprovisioningImageReconciler) AddIronicAgentToInfraEnv(ctx context.Co
 		return false, err
 	}
 
+	mirrorRegistryConfiguration, err := mirrorregistries.ProcessMirrorRegistryConfig(ctx, log, r.Client, infraEnv.Spec.MirrorRegistryRef)
```

Comment:
This function is only intended to set the internal override. We shouldn't need to also consider something the infraenv controller is already doing here. It's safe to remove this.